crates/fluvio-spu/src/services/public/produce_handler.rs

```diff
@@ -51,12 +53,14 @@ pub async fn handle_produce_request(
     request: RequestMessage<DefaultProduceRequest>,
     ctx: DefaultSharedGlobalContext,
 ) -> Result<ResponseMessage<ProduceResponse>, Error> {
+    let metrics = super::metrics::StorageMetrics::new();
+
     let (header, produce_request) = request.get_header_request();
     trace!("Handling ProduceRequest: {:#?}", produce_request);
 
     let mut topic_results = Vec::with_capacity(produce_request.topics.len());
     for topic_request in produce_request.topics.into_iter() {
-        let topic_result = handle_produce_topic(&ctx, topic_request).await;
+        let topic_result = handle_produce_topic(&ctx, topic_request, &metrics).await;
```

Comment:
Instead of passing as argument, we can add to context.  not show stopper

Reply:
done